x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -567,6 +574,7 @@ public void getAutodetectParams(Job job, Consumer<AutodetectParams> consumer, Co
                                 MultiSearchResponse.Item itemResponse = response.getResponses()[i];
                                 if (itemResponse.isFailure()) {
                                     errorHandler.accept(itemResponse.getFailure());
+                                    return;
```

Comment:
If we opt to add `return`s when we signal a failure, then we might as well use that to simplify the code and remove those `else`s. Same holds for the added `return`s a bit lower.